### PR TITLE
Update GetIntent adapter to 1.0 version

### PR DESCRIPTION
**Pull request by dhargo**
## Type of change

- [x] Other

## Description of change

GetIntent adapter for **Prebid.js 1.0** version

- test parameters for validating bids
```
{
  bidder: 'getintent',
  params: {
    pid: '7',
    tid: 'test01',
    // ...
  }
}
```

- contact email of the adapter’s maintainer
server-dev@getintent.com
- [x] official adapter submission

### CODE REVIEW

File: modules/getintentBidAdapter.js [hunk 1]
```diff
@@ -1,78 +1,130 @@
...
+
+  /**
+   * Callback for bids, after the call to DSP completes.
+   * Parse the response from the server into a list of bids.
+   *
+   * @param {object} serverResponse A response from the server.
+   * @param {BidRequest} bidRequest
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidRequest) {
```
Comment (matthewlane) [inline modules/getintentBidAdapter.js]: #1748 changed the first argument of `interpretResponse` to:

```JavaScript
{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
```

so doing something like

```JavaScript
serverResponse = serverResponse.body
```

(or however you'd prefer) in this function should give you the expected behavior

File: modules/getintentBidAdapter.js [hunk 2]
```diff
@@ -1,78 +1,130 @@
...
+    const bids = [];
+    if (serverResponse && serverResponse.no_bid !== 1) {
+      let size = parseSize(serverResponse.size);
+      let bid = {
+        requestId: bidRequest.bidId,
+        bidderCode: spec.code,
```
Comment (matthewlane) [inline modules/getintentBidAdapter.js]: `bidderCode` can be dropped from the bid response object, it'll be set automatically by `bidderFactory`. Docs just updated, sorry for the confusion

File: modules/getintentBidAdapter.js [hunk 3]
```diff
@@ -1,78 +1,130 @@
...
+  let giBidRequest = {
+    pid: bidRequest.params.pid, // required
+    tid: bidRequest.params.tid, // required
+    known: bidRequest.params.known || 1,
+    is_video: bidRequest.mediaType === 'video',
+    resp_type: 'JSON'
+  };
+  if (bidRequest.sizes) {
+    // TODO: add support for multiple sizes
+    giBidRequest.size = bidRequest.sizes[0].join('x');
```
Comment (matthewlane) [inline modules/getintentBidAdapter.js]: Adunit sizes may be defined as `[300, 250]`, in which case `sizes[0]` would be `300` and cause a `TypeError`. Be sure to check that `bidRequest.sizes[0]` is an array before calling `join`

File: modules/getintentBidAdapter.js [hunk 4]
```diff
@@ -1,78 +1,130 @@
...
+        bidderCode: spec.code,
+        cpm: serverResponse.cpm,
+        width: size[0],
+        height: size[1],
+        mediaType: bidRequest.mediaType || 'banner'
+      };
+      if (bidRequest.creative_id) bid.creativeId = bidRequest.creative_id;
+      if (bidRequest.mediaType === 'video') {
+        bid.vastUrl = serverResponse.vast_url;
+        bid.descriptionUrl = serverResponse.vast_url;
```
Comment (matthewlane) [inline modules/getintentBidAdapter.js]: `bid.descriptionUrl` is no longer required, `bid.vastUrl` alone will work now

File: modules/getintentBidAdapter.js [hunk 5]
```diff
@@ -1,78 +1,130 @@
...
+   *
+   * @param {object} serverResponse A response from the server.
+   * @param {BidRequest} bidRequest
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidRequest) {
+    const bids = [];
+    if (serverResponse && serverResponse.no_bid !== 1) {
+      let size = parseSize(serverResponse.size);
+      let bid = {
```
Comment (matthewlane) [inline modules/getintentBidAdapter.js]: A few of the required bid response fields are missing: `ttl`, `netRevenue`, and `currency`. See the table in this section http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response for descriptions and examples of each of these fields

**Comment by dhargo** [comment]
@matthewlane Thanks for the comments. I've commited fixes. Here is new parameters to use for testing bid response
```
http://px.adhigh.net/rtb/direct_banner?pid=7&tid=test01&known=1&is_video=false&resp_type=JSON&size=300x250& 
```

File: modules/getintentBidAdapter.js [hunk 6]
```diff
@@ -1,78 +1,142 @@
...
+   * @param {BidRequest[]} bidRequests - an array of bids.
+   * @return ServerRequest[]
+   */
+  buildRequests: function(bidRequests) {
+    return bidRequests.map(bidRequest => {
+      let giBidRequest = buildGiBidRequest(bidRequest);
+      return {
+        method: 'GET',
+        url: buildUrl(giBidRequest),
+        data: giBidRequest,
```
Comment (matthewlane) [inline modules/getintentBidAdapter.js]: This should potentially return the `bidRequest.bidId` so that the `interpretResponse` function has access to it

**Comment by dhargo** [comment]
@matthewlane Thank you for the comments. Fixed missing fields (`requestId`, `creativeId`).

Have to notice: method signature `interpretResponse: function(serverResponse, request)` and example in documentation http://prebid.org/dev-docs/bidder-adapter-1.html looks like we always have BidRequest (`request` parameter) when interpreting server bid response; so it is unclear why do we have to pass `requestId` to server and back.

**Comment by matthewlane** [comment]
Sorry for the confusion about that. The second parameter to `interpretResponse` is the bid request object which is available for use if needed. This object is whatever was returned the `buildRequests` function. Bid request and response objects need to have matching ids and one way of doing that is returning the id from `buildRequests`, but they can also be sent and received from the server as you've done here. In any case I was able to verify bids with your adapter so this is good for merge

**Pull request merged**